### PR TITLE
[BOP-1035] MetalLB

### PR DESCRIPTION
**Pull request by sakshisharma84**

### CODE REVIEW

File: content/docs/operations/metallb/_index.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+spec:
+addresses:
+- 192.168.1.240-192.168.1.250
+---
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+name: example
+namespace: metallb-system
+```
```
Comment (quadespresso) [inline content/docs/operations/metallb/_index.md:123]: 1. Shouldn't the YAML doc also begin with `---`?
2. Overall formatting appears to be incorrect.

Suggested revision:

```yaml
---
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
  name: first-pool
  namespace: metallb-system
spec:
  addresses:
  - 192.168.1.240-192.168.1.250
---
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: example
  namespace: metallb-system
```

File: content/docs/operations/metallb/_index.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+### Managing IP address pools during upgrade
+
+During the upgrade, if metalLB is enabled in MKE 3, the configured IP address pool details are displayed in upgrade summary.
+
+#### Example
+
+MetalLB enabled in MKE 3 with the following configuration.
+
+```yaml
```
Comment (quadespresso) [inline content/docs/operations/metallb/_index.md:135]: Looks like this should be `toml` and not `yaml`.

**Review by quadespresso — CHANGES_REQUESTED** [review]

File: content/docs/operations/metallb/_index.md (hunk 1 above)
Comment (sakshisharma84) [reply content/docs/operations/metallb/_index.md:123]: @quadespresso 

> Shouldn't the YAML doc also begin with ---?
Not necessarily. It works even w/o it. And if it looks better in the docs, I can definitely add it.

Comment (sakshisharma84) [reply content/docs/operations/metallb/_index.md]: done!

File: content/docs/operations/metallb/_index.md (hunk 2 above)
Comment (sakshisharma84) [reply content/docs/operations/metallb/_index.md]: done!

File: content/docs/operations/metallb/_index.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+
+
+
+## MKE version comparison: Metallb configuration parameters
+
+| MKE-3                                                          | MKE-4                                   |
+|----------------------------------------------------------------|-----------------------------------------|
+| [cluster_config.metallb_config.enabled]                        | addons.metallb.enabled                  |
+| [[cluster_config.metallb_config.metallb_ip_addr_pool]]         | `Deprecated`                            |
```
Comment (quadespresso) [inline content/docs/operations/metallb/_index.md:93]: Apologies for the nits being picked here, but I'm wondering whether each of the table values should be formatted as code...?

eg,
```
| MKE-3                                                          | MKE-4                                   |
|----------------------------------------------------------------|-----------------------------------------|
| `[cluster_config.metallb_config.enabled]`                      | `addons.metallb.enabled`                  |
| `[[cluster_config.metallb_config.metallb_ip_addr_pool]]`       | `Deprecated`  
```

Also, could I get clarification where `addons.metallb.enabled` is located in the output of `mkectl init`?

If from this block:
```yaml
spec:
  addons:
    - chart:
        name: metallb
        repo: https://metallb.github.io/metallb
        values: |
          controller:
            tolerations:
              - key: node-role.kubernetes.io/master
                operator: Exists
                effect: NoSchedule
          speaker:
            frr:
              enabled: false
        version: 0.14.7
      dryRun: false
      enabled: false
      kind: chart
      name: metallb
      namespace: metallb-system
```

then wouldn't the JSONpath be `addons[0].enabled` or more precisely `.spec.addons[0].enabled`?

I had a quick go with a few alternatives; nothing seemed to align with `addons.metallb.enabled`:

```bash
❯ mkectl init | yq 'addons'
Error: 1:1: invalid input text "addons"

❯ mkectl init | yq '.addons'
null

❯ mkectl init | yq '.spec.addons'
- chart:
    name: metallb
    repo: https://metallb.github.io/metallb
    values: |
      controller:
          tolerations:
              - key: node-role.kubernetes.io/master
                operator: Exists
                effect: NoSchedule
      speaker:
          frr:
              enabled: false
    version: 0.14.7
  dryRun: false
  enabled: false
  kind: chart
  name: metallb
  namespace: metallb-system

❯ mkectl init | yq '.spec.addons.metal'
Error: cannot index array with 'metal' (strconv.ParseInt: parsing "metal": invalid syntax)

❯ mkectl init | yq '.spec.addons[0].chart.name'
metallb

❯ mkectl init | yq '.spec.addons[0].enabled'
false
```

File: content/docs/operations/metallb/_index.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+| namespace                             | Indicates the namespace used for deploying metallb        | metallb-system                                                                            |
+| chart.name                            | Indicates the name of the metallb helm chart              | metallb                                                                                   |
+| chart.repo                            | Indicates the metallb helm repository                     | https://metallb.github.io/metallb                                                         |
+| chart.version                         | Indicates the version of the metallb helm chart           | 0.14.7                                                                                    |
+| chart.values.controller.tolerations   | Indicates the tolerations for the metallb controller pod  | - key: node-role.kubernetes.io/master<br><br/>operator: Exists <br><br/>effect: NoSchedule|
+| chart.values.speaker.frr.enabled      | Indicates whether FRR is enabled for metallb speaker      | false                                                                                     |
```
Comment (quadespresso) [inline content/docs/operations/metallb/_index.md:83]: I think columns 1 and 3 should have their values formatted as code. All rows, except for the one item I called out elsewhere as needing a `yaml` code fence.

**Review by quadespresso — CHANGES_REQUESTED** [review]

File: content/docs/operations/metallb/_index.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+| name                                  | Indicates the name of the addon                           | metallb                                                                                   |
+| namespace                             | Indicates the namespace used for deploying metallb        | metallb-system                                                                            |
+| chart.name                            | Indicates the name of the metallb helm chart              | metallb                                                                                   |
+| chart.repo                            | Indicates the metallb helm repository                     | https://metallb.github.io/metallb                                                         |
+| chart.version                         | Indicates the version of the metallb helm chart           | 0.14.7                                                                                    |
+| chart.values.controller.tolerations   | Indicates the tolerations for the metallb controller pod  | - key: node-role.kubernetes.io/master<br><br/>operator: Exists <br><br/>effect: NoSchedule|
```
Comment (quadespresso) [inline content/docs/operations/metallb/_index.md:82]: For just this row, the third column should probably be reformatted as `yaml` code fence, ie,

    ```yaml
    - key: node-role.kubernetes.io/master
      operator: Exists
      effect: NoSchedule
    ```

**Review by quadespresso — DISMISSED** [review]
Missed this one last night.

**Review by KoryKessel-Mirantis — CHANGES_REQUESTED** [review]
First Docs review round for MetalLB content. NFD to follow.

**Comment by ry4nz** [comment]
@quadespresso @KoryKessel-Mirantis Please review again.

**Comment by ry4nz** [comment]
@KoryKessel-Mirantis Let's merge as is then you may work on the formatting later?

**Review by KoryKessel-Mirantis — APPROVED** [review]

File: content/docs/operations/metallb/_index.md (hunk 4 above)
Comment (KoryKessel-Mirantis) [reply content/docs/operations/metallb/_index.md]: @quadespresso, for the second-to-last column 3, I am going to need guidance...guessing this is the so-called "yaml code fende"?

**Pull request merged**